### PR TITLE
[Concept] Add public gateway type - authenticated gateway

### PR DESCRIPTION
**Pull request by zikunfan**

### CODE REVIEW

**Comment by zikunfan** [comment]
@johnnymatthews Could you please take a look at this PR?

File: docs/concepts/ipfs-gateway.md [hunk 1]
```diff
@@ -70,6 +71,16 @@ Choosing the form of gateway usage has security, performance, and other function
 
 The examples discussed in the earlier sections above illustrated the use of read-only HTTP gateways to fetch content from IPFS via an HTTP GET method. _Writeable_ HTTP gateways also support `POST`, `PUT`, and `DELETE` methods.
 
+### Authenticated gateways
+
+If the gateway providers want to limit the access only to the requests with authentication, they may need to config a reverse proxy, develop a IPFS plugin or set a cache-layer above IPFS.
```
Comment (johnnymatthews) [inline docs/concepts/ipfs-gateway.md:76]:
```suggestion
If a gateway provider wants to limit access to requests with authentication, they may need to configure a reverse proxy, develop an IPFS plugin, or set a cache-layer above IPFS.
```

File: docs/concepts/ipfs-gateway.md [hunk 2]
```diff
@@ -70,6 +71,16 @@ Choosing the form of gateway usage has security, performance, and other function
...
+
+Reverse proxy is the most popular way for providers handling authentication. This tutorial configuring private gateway includes a description of controling access with Nginx. Reverse proxy can also keep the original IPFS API calls which makes gateway adaptable to all IPFS SDK/toolkits.
```
Comment (johnnymatthews) [inline docs/concepts/ipfs-gateway.md:78]:
```suggestion
Configuring a reverse proxy is the most popular way for providers handling authentication. Reverse proxy can also keep the original IPFS API calls which makes gateway adaptable to all IPFS SDK and toolkits.
```

File: docs/concepts/ipfs-gateway.md [hunk 3]
```diff
@@ -70,6 +71,16 @@ Choosing the form of gateway usage has security, performance, and other function
...
+
+![Auth with Reverse proxy](./images/ipfs-gateways/public-authed-gateway.png)
+
+Providers can design their own centralized auth service(like [Infura IPFS Auth](https://infura.io/docs/ipfs#section/Authentication/Overview)), or a decentralized auth service(like [IPFS W3Auth](https://wiki.crust.network/docs/en/buildIPFSWeb3AuthGW)).
```
Comment (johnnymatthews) [inline docs/concepts/ipfs-gateway.md:82]:
```suggestion
Providers can design their own centralized authentication service like [Infura IPFS Auth](https://infura.io/docs/ipfs#section/Authentication/Overview), or a decentralized authentication service like [IPFS W3Auth](https://wiki.crust.network/docs/en/buildIPFSWeb3AuthGW)).
```

**Review by johnnymatthews — CHANGES_REQUESTED** [review]
Here's a couple of minor suggestions. This is good to merge when they've been added.

**Comment by zikunfan** [comment]
@johnnymatthews Comments already resolved

File: docs/concepts/ipfs-gateway.md [hunk 4]
```diff
@@ -73,13 +73,13 @@ The examples discussed in the earlier sections above illustrated the use of read
 
 ### Authenticated gateways
 
-If the gateway providers want to limit the access only to the requests with authentication, they may need to config a reverse proxy, develop a IPFS plugin or set a cache-layer above IPFS.
+If a gateway provider wants to limit access to requests with authentication, they may need to configure a reverse proxy, develop an IPFS plugin, or set a cache-layer above IPFS.
 
-Reverse proxy is the most popular way for providers handling authentication. This tutorial configuring private gateway includes a description of controling access with Nginx. Reverse proxy can also keep the original IPFS API calls which makes gateway adaptable to all IPFS SDK/toolkits.
+Configuring a reverse proxy is the most popular way for providers handling authentication. Reverse proxy can also keep the original IPFS API calls which makes gateway adaptable to all IPFS SDK and toolkits.
```
Comment (johnnymatthews) [inline docs/concepts/ipfs-gateway.md:78]:
```suggestion
Configuring a reverse proxy is the most popular way for providers handling authentication. A reverse proxy can also keep the original IPFS API calls which makes gateway adaptable to all IPFS SDK and toolkits.
```

**Review by johnnymatthews — APPROVED** [review]

**Pull request merged**